toolkit/tools/scheduler/schedulerutils/graphbuildstate.go

```diff
@@ -145,7 +145,8 @@ func (g *GraphBuildState) RecordBuildResult(res *BuildResult) {
 		g.nodeToState[node] = state
 	}
 
-	if !res.Skipped && !res.UsedCache {
+	logger.Log.Debugf("Recording build result: %s", res.Node.FriendlyName())
+	if !allowToolchainRebuilds && !res.Skipped && !res.UsedCache {
```

Comment:
I've noticed this just now: let's also update the log in the `else` part, since now we have one more reason to skip this check, so the message will be incorrect in case of an allowed toolchain rebuild.